gempy/core/data.py

```diff
@@ -281,76 +281,90 @@ class Colors:
     """
     Object that handles the color management in the model.
     """
-
     def __init__(self, surfaces):
         self.surfaces = surfaces
-
-    def generate_colordict(self, hex_colors='palettes',
-                           palettes: List[str] = 'default', out=False):
-        """generate colordict that assigns black to faults and random colors to surfaces
+        self.colordict = None
+        self._hexcolors_soft = [
+            '#015482', '#9f0052', '#ffbe00', '#728f02', '#443988',
+            '#ff3f20', '#5DA629', '#b271d0', '#72e54a', '#583bd1',
+            '#d0e63d', '#b949e2', '#95ce4b', '#6d2b9f', '#60eb91',
+            '#d746be', '#52a22e', '#5e63d8', '#e5c339', '#371970',
+            '#d3dc76', '#4d478e', '#43b665', '#d14897', '#59e5b8',
+            '#e5421d', '#62dedb', '#df344e', '#9ce4a9', '#d94077',
+            '#99c573', '#842f74', '#578131', '#708de7', '#df872f',
+            '#5a73b1', '#ab912b', '#321f4d', '#e4bd7c', '#142932',
+            '#cd4f30', '#69aedd', '#892a23', '#aad6de', '#5c1a34',
+            '#cfddb4', '#381d29', '#5da37c', '#d8676e', '#52a2a3',
+            '#9b405c', '#346542', '#de91c9', '#555719', '#bbaed6',
+            '#945624', '#517c91', '#de8a68', '#3c4b64', '#9d8a4d',
+            '#825f7e', '#2c3821', '#ddadaa', '#5e3524', '#a3a68e',
+            '#a2706b', '#686d56'
+        ]  # source: https://medialab.github.io/iwanthue/
+
+    def generate_colordict(
+            self,
+            hex_colors: Union[List[str], str] = 'palettes',
+            palettes: List[str] = 'default',
+    ):
+        """Generates and sets color dictionary.
 
         Args:
-           hex_colors (list[str], str): List of hex values or specific values. In the
-            future this could accommodate the actual geological palettes. For example
-            striplog has a quite good set of palettes.
+           hex_colors (list[str], str): List of hex color values. In the future this could
+           accommodate the actual geological palettes. For example striplog has a quite
+           good set of palettes.
                 * palettes: If hexcolors='palettes' the colors will be chosen from the
                    palettes arg
                 * soft: https://medialab.github.io/iwanthue/
-
-           palettes (list[str]): list with name of seaborn palettes
+           palettes (list[str], optional): list with name of seaborn palettes. Defaults to 'default'.
         """
-        import seaborn as sns
+        import seaborn as sns  # should this be here? we need colors every time anyways
```

Comment:
Indeed this is a really weird place for an import. I don't think it should be here.

At least it can just go to the top of this file. Also maybe just `import seaborn` and refactor instead of `import seaborn as sns`.

Reply:
`sns` is pretty much standard import jargon for seaborn, I'll move it up though